signal.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2013-2014 The btcsuite developers
```

Comment:
Rather than adding this file twice, are we able to reuse `signal.go` from #528?

Reply:
My thinking is that they may diverge at some point.
I guess that will work just as well tho.